templates/var/lib/ansible/playbooks/main.yml

```diff
@@ -40,14 +40,40 @@
   - name: Restart node service
     service: name={{openshift.common.service_type}}-node state=restarted
 
+<%#deploy_registry%>
+- name: Waiting for registry setup
+  hosts: masters[0]
+  tasks:
+  - action: shell oc get pod | grep -v deploy | grep registry | grep Running
+    register: result
+    until: result.rc == 0
+    retries: 5
+    delay: 60
+<%/deploy_registry%>
+
+<%#deploy_router%>
+- name: Waiting for router setup
+  hosts: masters[0]
+  tasks:
+  - action: shell oc get pod | grep -v deploy | grep router | grep Running
+    register: result
+    until: result.rc == 0
+    retries: 5
```

Comment:
Would it make sense to do at least 10 or 15 retries? I was quite far to get into this 5m limit. It took about 13 mins for me to fetch and start the image.